docs/extra/optimize_function.rst

```diff
@@ -0,0 +1,297 @@
+functions if you do not need to take advantage of Sherpa's model and
+data classes.
+
+The optimizers in :py:mod:`sherpa.optmethods.optfcts` are functions
+which follow the same interface::
+
+  optimizer(cb, start, lolims, hilims, **kwargs)
+
+where `cb` is a callable that given a set or parameters returns the
+current statistic, `start` is the starting positions of the
+parameters, `lolims` and `hilims` are the bounds of the parameters,
+and the remaining keyword arguments are specific to the optimizer.
+
+Examples
+========
+
+The following imports have been made::
+
+  >>> import numpy as np
+  >>> from matplotlib import pyplot as plt
+  >>> from matplotlib import colors
+  >>> from mpl_toolkits.mplot3d import Axes3D
+  >>> from astropy.table import Table
+  >>> from sherpa.optmethods import optfcts
+
+A simple function
+-----------------
+
+The function we want to optimize is a version of the
+`Rosenbrock function <https://en.wikipedia.org/wiki/Rosenbrock_function>`_,
+where `a=2` and `b=10`, so the best-fit location is at::
+
+  x = 2
+  y = 2 * 2
+
+.. note::
+   The Sherpa test suite contains a number of test cases based on
+   functions like this, taken from
+   `More, J. J., Garbow, B. S., and Hillstrom, K. E.. Testing unconstrained optimization software. United States: N. p., 1978. Web. doi:10.2172/6650344. <https://www.osti.gov/biblio/6650344>`_.
+
+The function is::
+
+  def rosenbrock(x, y):
+      a, b = 2, 10
+      return (a - x) * (a - x) + b * (y - x * x) * (y - x * x)
+
+We can look at the surface close to the best-fit location
+What does the function look like?
+
+  >>> y, x = np.mgrid[-2:5.1:0.1, -3:3.1:0.1]
+  >>> surface = rosenbrock(x, y)
+  >>> print(surface.shape)
+  (71, 61)
+  >>> print(surface.min())
+  0.0
+  >>> print(surface.max())
+  1235.0
+
+This can be visualized as::
+
+  >>> fig, ax = plt.subplots(subplot_kw={"projection": "3d"})
+  >>> surf = ax.plot_surface(x, y, surface,
+  ...                        cmap=cm.plasma_r, alpha=0.8,
+  ...                        vlim=0, vmax=300)
+  >>> ax.set_xlabel('x')
+  >>> ax.set_ylabel('y')
+  >>> ax.view_init(30, -100)
+
+.. image:: ../_static/extra/rosenbrock-surface.png
+
+In order to optimize this function we need to determine how we
+want to define the "search surface"; that is the value that the
+optimizer is going to try and minimize. For this dataset, where
+the minimum is 0, we can use the square of the function (i.e. a
+least-squares model). Now, the function we pass to the minimizer
+requires two values - the "statistic" and a per-parameter breakdown
+of the parameter - but for now we are going to ignore the latter,
+as it's only needed for some optimizers. This gives us::
+
+  def to_optimize1(args):
+      x = args[0]
+      y = args[1]
+      pa = (2 - x) * (2 - x)
+      pb = 10 * (y - x * x) * (y - x * x)
+      stat = pa * pa + pb * pb
+      return stat, None
+
+The function can tehn be monimized by passing the routine, the
```

Comment:
```suggestion
The function can then be minimized by passing the routine, the
```
I know you put that in jsut ot maek suer we pya attentions.

Reply:
Look. You thought we were learning about how to speed up code, not monomize functions!

Reply:
monomized, sure. But monimized? https://in.pinterest.com/srsirisharao/monomize/